src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java

```diff
@@ -102,12 +103,27 @@ public void save( final URI identifier, final T element ) throws IOException {
 		}
 	}
 
-	@SuppressWarnings( "unchecked" )
+	public void save( final OutputStream outputStream, final T element ) throws IOException {
+		final Output output = new Output( new LZ4FrameOutputStream( outputStream ) );
+		output.writeString( version );
+		kryo.writeClassAndObject( output, element );
+		output.close();
```

Comment:
This output will not be closed on error, see above.

Reply:
On error, this is no problem, since the resulting file would be corrupt anyways. This close is not there to prevent memory leaks (which most often is the result of forgetting to call close), but to prevent a corrupt result file.